### PR TITLE
Issue #9502: update example of AST for TokenTypes.LITERAL_TRUE

### LINKED ISSUE #9502: update example of AST for TokenTypes.LITERAL_TRUE
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRUE

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by MahmoudAbdelazim**
Closes #9502
image
Code used for generating example:
```
mahmoud:/data/checkstyle/checkstyle test$ more Test.java 

public class A {
    public void b {
    	boolean a = true;
    }
}
```
Output:
```
mahmoud:/data/checkstyle/checkstyle test$ java -jar checkstyle-8.41-all.jar -t Test.java 
CLASS_DEF -> CLASS_DEF [2:0]
|--MODIFIERS -> MODIFIERS [2:0]
|   `--LITERAL_PUBLIC -> public [2:0]
|--LITERAL_CLASS -> class [2:7]
|--IDENT -> A [2:13]
`--OBJBLOCK -> OBJBLOCK [2:15]
    |--LCURLY -> { [2:15]
    |--VARIABLE_DEF -> VARIABLE_DEF [3:4]
    |   |--MODIFIERS -> MODIFIERS [3:4]
    |   |   `--LITERAL_PUBLIC -> public [3:4]
    |   |--TYPE -> TYPE [3:11]
    |   |   `--LITERAL_VOID -> void [3:11]
    |   `--IDENT -> b [3:16]
    |--INSTANCE_INIT -> INSTANCE_INIT [3:18]
    |   `--SLIST -> { [3:18]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [4:5]
    |       |   |--MODIFIERS -> MODIFIERS [4:5]
    |       |   |--TYPE -> TYPE [4:5]
    |       |   |   `--LITERAL_BOOLEAN -> boolean [4:5]
    |       |   |--IDENT -> a [4:13]
    |       |   `--ASSIGN -> = [4:15]
    |       |       `--EXPR -> EXPR [4:17]
    |       |           `--LITERAL_TRUE -> true [4:17]
    |       |--SEMI -> ; [4:21]
    |       `--RCURLY -> } [5:4]
    `--RCURLY -> } [6:0]
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -3088,6 +3088,23 @@
...
+     * <pre>
+     * VARIABLE_DEF -&gt; VARIABLE_DEF
+     *  |--MODIFIERS -&gt; MODIFIERS
+     *  |--TYPE -&gt; TYPE
+     *  |   `--LITERAL_BOOLEAN -&gt; boolean
+     *  |--IDENT -&gt; a
+     *  `--ASSIGN -&gt; =
+     *      `--EXPR -&gt; EXPR
+     *          `--LITERAL_TRUE -&gt; true
+     * SEMI -&gt; ;
```
Comment (sabaka) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3105]: Hanging semi doesn't look good. Should we also include connection between variable_def and semi?

**Review by sabaka — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (MahmoudAbdelazim) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:3105]: Ok I'll do that.

Comment (MahmoudAbdelazim) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Done.

Comment (MahmoudAbdelazim) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Done.

**Review by romani — APPROVED** [review]

**Pull request merged**